### PR TITLE
STM32: Espruino Pico 

### PR DESCRIPTION
**Pull request by hierophect**
This PR adds support for the [Espruino Pico](https://www.espruino.com/Pico) board ([Adafruit Link](https://www.adafruit.com/product/2621)). This board normally runs a javascript interpreter and is updated via a bootloader over a web interface - because Circuitpython depends more on the internal filesystem, this bootloader is not supported and you MUST use the built in F4 ROM bootloader included on all ST chips. This process is documented in the boards/espruino_pico/README.md file.

### CODE REVIEW

**Comment by hierophect** [comment]
@tannewt need PID for this when you have a sec.

**Comment by ladyada** [comment]
```
Espruino Pico     VID 0x239A    PID 0x008D  # bootloader
                                PID 0x808D  # arduino
                                PID 0x808E  # circuitpython
```

File: ports/stm32f4/boards/STM32F401_fs.ld [hunk 1]
```diff
@@ -5,10 +5,10 @@
 /* Specify the memory areas */
 MEMORY
 {
-    FLASH (rx)      : ORIGIN = 0x08000000, LENGTH = 512K /* entire flash */
+    FLASH (rx)      : ORIGIN = 0x08000000, LENGTH = 384K /* entire flash */
     FLASH_ISR (rx)  : ORIGIN = 0x08000000, LENGTH = 16K /* sector 0 */
     FLASH_FS (rx)   : ORIGIN = 0x08004000, LENGTH = 48K /* sectors 1,2,3 are 16K */
-    FLASH_TEXT (rx) : ORIGIN = 0x08010000, LENGTH = 448K /* sector 4 is 64K, sectors 5,6,7 are 128K */
+    FLASH_TEXT (rx) : ORIGIN = 0x08010000, LENGTH = 320K /* sector 4 is 64K, sectors 5,6,7 are 128K */
```
Comment (tannewt) [inline ports/stm32f4/boards/STM32F401_fs.ld]: Why change this? It doesn't look like the Pico uses it but something else may.

Comment (hierophect) [reply ports/stm32f4/boards/STM32F401_fs.ld]: I just mixed up my linker files like a chucklehead, thanks for catching it

Comment (hierophect) [reply ports/stm32f4/boards/STM32F401_fs.ld]: renamed them so they're more clear too

**Comment by hierophect** [comment]
@tannewt the bootloader always takes at least 16KB (one sector) because it can't overwrite itself. It might take even more in the Espruino case, it was kind of hard to tell from their code. It also isn't super well documented how to upload your own binaries with it, since it's designed for use with their online IDE. I don't think it'd be any easier for new users to use it rather than the DFU option, as compared to a UF2 bootloader.

**Comment by hierophect** [comment]
I'm not an Espruino pro so I could be wrong, but if any come along and think there's a way to get around this, then it'd be easily revisited.

**Comment by ladyada** [comment]
@hierophect and/or @makermelissa - add these to circuitpy.org next please :)

**Comment by makermelissa** [comment]
Done. PR is https://github.com/adafruit/circuitpython-org/pull/404

**Comment by ladyada** [comment]
did a lil hardware test - just checks the button and LEDs
```
import time
import board
from digitalio import DigitalInOut, Direction, Pull

led1 = DigitalInOut(board.LED1)
led1.direction = Direction.OUTPUT
led2 = DigitalInOut(board.LED2)
led2.direction = Direction.OUTPUT
switch = DigitalInOut(board.BTN)
switch.direction = Direction.INPUT
switch.pull = Pull.DOWN

while True:
    if switch.value:
        led1.value = False
        led2.value = True
    else:
        led1.value = True
        led2.value = False

    time.sleep(0.01)  # debounce delay
```

**Pull request merged**